core/src/main/scala/org/apache/spark/api/python/StreamingPythonRunner.scala

```diff
@@ -109,7 +109,10 @@ private[spark] class StreamingPythonRunner(
    */
   def stop(): Unit = {
     pythonWorker.foreach { worker =>
-      SparkEnv.get.destroyPythonWorker(pythonExec, workerModule, envVars.asScala.toMap, worker)
+      pythonWorkerFactory.foreach { factory =>
```

Comment:
PTAL @WweiL & @bogao007. I have manually tested. Even with the NPE issue that Bo mentioned, the process is getting removed.